software/controller/test_libs/mocks/mock_stepper.h

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2022, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include "mock_spi.h"
+#include "stepper.h"
+
+namespace StepMotor {
+
+enum class MoveDir { Shortest, Negative, Positive };
+
+// Class that allows us to (very approximately) emulate a stepper motor
+class TestStepper {
```

Comment:
I don't think I fully understand how this mocks the Stepper class, as it doesn't specialize some abstract virtual interface. How does it relate to `Handler` that is in `stepper.h`?

Reply:
Ok, so it is no mocking the main class that is defined in `stepper.h` that's why the confusion. I don't think we necessarily need to mock that handler class (though it may be helpful in writing tests for PinchValve?). But at least we should then maybe call it something like `StepperEmulator` or `PowerStepEmulator`? Even if it's not a complete "simluator" of everything that such a system is? Not sure what the best word for it is, really.